core/kernel/dt.c

```diff
@@ -4,6 +4,7 @@
  */
 
 #include <assert.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
```

Comment:
This makes this file ARM-specific which is undesirable, especially now that we know that some people are porting OP-TEE to other architecture(s). Perhaps `dt_get_irq()` should remain declared in `core/include/kernel/dt.h` but its implementation should be moved under `core/arch/arm/kernel`?

Reply:
hi Jerome

like dt-bindings is specific by drivers, I could proposed you a second possibility:
- move the dt_get_irq in interrupt framework.
- Add a xlate function in itr_chip stucture, which allow to translate the specific properties of driver.
- this xlate function will be initialize by driver while init step
example: 
```
 void gic_init_base_addr(struct gic_data *gd, vaddr_t gicc_base __maybe_unused,
 			vaddr_t gicd_base)
 {
 	gd->gicc_base = gicc_base;
 	gd->gicd_base = gicd_base;
 	gd->max_it = probe_max_it(gicc_base, gicd_base);
 	gd->chip.ops = &gic_ops;
+#ifdef CFG_DT
+	gd->chip.xlate = gic_xlate;
+#endif
 }
```

-the dt_get_irq function find the interrupt property then call the dedicated xlate function of interrupt driver
example:
```
+#ifdef CFG_DT
+int dt_get_irq(const void *fdt, int node)
+{
+	const uint32_t *prop = NULL;
+	int len = 0;
+	int it_num = DT_INFO_INVALID_INTERRUPT;
+
+	if (!itr_chip || !itr_chip->xlate)
+		return it_num;
+
+	prop = fdt_getprop(fdt, node, "interrupts", &len);
+	if (!prop)
+		return it_num;
+
+	return itr_chip->xlate(prop, len);
+}
+#endif
```

what do you think about it ?
I will pull a fixup to show this proposal

Reply:
That would work, but is it useful? `xlate()` would be attached to `itr_chip` and there is only one instance of `itr_chip` in `core/kernel/interrupt.c`, so what does this bring us compared to a global function defined in the interrupt controller driver?

Reply:
this allow to not duplicate the first part of code find the node "Interrupt", and translate the specific part in the driver.
If there are several architectures, the first part will be the same, but the second will be in own driver.

Reply:
> this allow to not duplicate the first part of code find the node "Interrupt",

OK, but that's just one line of code isn't it? (`fdt_getprop()`). I am just trying to avoid over-engineering.

Reply:
Yes, that allow to split the responsibilities. the framework manage the property name "interrupt" and evolution (defined by dt-bindings or system device tree and which is not depend of cpu architecture), and avoid to define a dt_get_irq by architectures.
The drivers can focus only on translate parameters.
if you would, I could pull a fixup to show the small modification.

Reply:
OK, let's see the code.

Reply:
Rather than a fixup commit which removes this line I would prefer to see a clean series of patches, for instance in another PR. It is the best way to understand what is going on when reviewing.

Reply:
ok, I will push a new PR. I come from the linux world. I don't know the procedures for optee yet. :-)

Reply:
No worries -- in general we prefer fixups like you did (especially for minor fixes) but sometimes it's better to rework the patch series.